modules/in_band_dtmf/in_band_dtmf.c

```diff
@@ -0,0 +1,337 @@
+/**
+ * @file in_band_dtmf.c  DTMF decoder
+ */
+#include <time.h>
+#include <ctype.h>
+#include <string.h>
+#include <re.h>
+#include <rem.h>
+#include <baresip.h>
+
+
+/**
+ * @defgroup in_band_dtmf in_band_dtmf
+ *
+ * Audio filter that de- and encodes DTMF tones.
+ *
+ * The tone length of an encoded tone is 0.1s.
+ * New tones can be added while encoding is active.
+ */
+
+
+struct in_band_dtmf_filt_dec {
+	struct aufilt_enc_st af;  /* inheritance */
+	struct dtmf_dec *dec;
+	struct ua *ua;
+	struct call *call;
+	struct le le_priv;
+};
+
+
+struct in_band_dtmf_filt_enc {
+	struct aufilt_enc_st af;  /* inheritance */
+	struct mbuf *mb;
+	unsigned srate;
+	struct le le_priv;
+};
+
+
+static struct list encs;
+static struct list decs;
```

Comment:
are these needed ?

Reply:
I need to pass some information to the related functions (`in_band_dtmf_send` and `in_band_dtmf_dec_handler`). The `in_band_dtmf_dec_handler` provides the user argument `arg` which could be an option. But i also need to pass the current `ua` and `call` argument, which i get within `ua_event_handler`.

I'm not sure if the list approach is the most easiest implementation. I copied the idea from `modules/mixausrc/mixausrc.c`.  I guess i could also invent an active call flag to my structs, and use then two static variables and omit the list.

Which way do you prefer?

Reply:
Is the `decs` state still needed ?

Reply:
Yes, since the dec also manage the `struct tmr tmr_dtmf_end`, used to delay the dtmf end event.